docs/docs/server/server.md

```diff
@@ -0,0 +1,195 @@
+```
+
+### Proxy Configuration
+By default, the server only trusts localhost proxies (`127.0.0.1` and `::1`). If you're running behind a proxy (like Nginx, load balancer, etc.), configure trusted proxy IPs using the `TRUSTED_PROXIES` environment variable:
+
+```bash
+TRUSTED_PROXIES=10.0.0.1,10.0.0.2
+```
+
+## API Endpoints
+
+For detailed information about available endpoints, see the [API Reference](../api/endpoints.md).
+
+## Agent Endpoints
+
+### Export Agent
+`GET /api/agent/export`
+
+Export an agent's configuration as JSON.
+
+**Parameters:**
+
+  - `agent-id` (required): The ID of the agent to export
+  - `version-id` (optional): The version ID to export (defaults to "development")
+
+**Response:** JSON configuration of the agent
+
+## Analytics Endpoints
+
+### Fetch Analytics
+`GET /api/analytics/fetch`
+
+Fetch analytics data for an agent.
+
+**Parameters:**
+
+- `agent-id` (required): The ID of the agent
+- `start-time` (optional): Start time for analytics data
+- `end-time` (optional): End time for analytics data
+- `limit` (optional): Maximum number of results (default: 100)
+- `analytics` (optional): Array of analytics types to fetch. If not specified, fetches all types:
+    `interactions`, `sessions`, `top_intents`, `top_slots`, `understood_messages`, `unique_users`, `token_usage`
+
+## Document Endpoints
+
+### Fetch Documents
+`GET /api/document/fetch`
+
+Fetch documents from a knowledge base.
+
+**Parameters:**
+
+- `page` (optional): Page number (default: 1)
+- `limit` (optional): Results per page (default: 10)
+- `document-type` (optional): Type of documents to fetch
+- `include-tags` (optional): Array of tags to include
+- `exclude-tags` (optional): Array of tags to exclude
+- `include-all-non-tagged` (optional): Include all non-tagged documents (default: false)
+- `include-all-tagged` (optional): Include all tagged documents (default: false)
+
+### Upload Document URL
+`POST /api/document/upload-url`
+
+Upload a document from a URL.
+
+**Request Body:**
+```json
+{
+  "url": "string (required)",
+  "name": "string (required)",
+  "overwrite": "boolean",
+  "max_chunk_size": "number",
+  "markdown_conversion": "boolean",
+  "llm_generated_q": "boolean",
+  "llm_prepend_context": "boolean",
+  "llm_based_chunking": "boolean",
+  "llm_content_summarization": "boolean",
+  "tags": "string[]"
+}
+```
+
+### Upload Document File
+`POST /api/document/upload-file`
+
+Upload a document file.
+
+**Form Data:**
+
+- `file` (required): The file to upload
+- `overwrite` (optional): Whether to overwrite existing document
+- `max_chunk_size` (optional): Maximum chunk size
+- `markdown_conversion` (optional): Convert to markdown
+- `llm_generated_q` (optional): Generate questions using LLM
+- `llm_prepend_context` (optional): Prepend context using LLM
+- `llm_based_chunking` (optional): Use LLM for chunking
+- `llm_content_summarization` (optional): Summarize content using LLM
+- `tags` (optional): Array of tags
+
```

Comment:
we should autogenerate this by generating the openapi spec: https://github.com/swaggo/gin-swagger